extensions/wikia/Chat2/js/emoticons.js

```diff
@@ -101,7 +101,7 @@ if(typeof EmoticonMapping === 'undefined'){
 					currentKey = url;
 					//$().log("  " + url + "...");
 				} else if (self._settings[currentKey]) {
-					var glyphMatch = emoticonArray[i].match(/^\*\*[ ]*([^*].*)/); // line starting with 2 "**"'s then optional spaces, then some non-empty content.
+					var glyphMatch = emoticonArray[i].match(/^\*\* *([^*"][^"]*)/); // line starting with 2 "**"'s then optional spaces, then some non-empty content.
```

Comment:
This basically stops matching content from the special emoticon page as soon as a ```"``` is encountered.  So for the example in the ticket:
```
(angry" onmouseover="alert(0);" "
```
it would match up to ```(angry``` as the glyph name and ignore everything else on the rest of the line.  If someone enters this in chat, the ```(angry``` part will trigger an emoticon to be drawn and the ```" onmouseover="alert(0);" "``` will be treated as chat text and escaped and printed.